stackstate_checks_base/tests/test_http_helper.py

```diff
@@ -0,0 +1,458 @@
+import unittest
+import json
+from schematics.models import Model, DataError
+from schematics.types import StringType, IntType, BooleanType
+from stackstate_checks.utils.http_helper import HTTPHelper, HTTPRequestType, HTTPAuthenticationType, HTTPResponseType
+from stackstate_checks.utils.http_helper import HTTPHelperRequestHandler, HTTPHelperSessionHandler, HTTPMethod
+from stackstate_checks.utils.http_helper import HTTPHelperResponseHandler, HTTPHelperConnectionHandler, \
+    HTTPHelperRequestModel, HTTPHelperConnectionModel, HTTPHelperSessionModel, HTTPHelperResponseModel
+
+"""
+Structures used within the test cases
+"""
+
+
+class BodySchematicTest(Model):
+    hello = IntType(required=True)
+
+
+class BodyRequestSchematicTest(Model):
+    hello = StringType(required=True)
+    pong = BooleanType(required=True)
+
+
+class BodyResponseSchematicTest(Model):
+    hello = StringType(required=True)
+    pong = BooleanType(required=True)
+
+
+"""
+ HTTP Helper Request Class
+"""
+
+
+class TestHTTPHelperRequestHandler(unittest.TestCase):
+    def test_get_request_model(self):
+        request_handler = HTTPHelperRequestHandler()
+        model = request_handler.get_request_model()
+        assert isinstance(model, HTTPHelperRequestModel)
+
+        # Test the base model. This should fail as a few things are required
+        try:
+            request_handler.validate_request_model()
+        except DataError as e:
+            assert len(e.errors) == 2
+
+    def test_create_request_object(self):
+        HTTPHelperRequestHandler({"method": "GET",
+                                  "endpoint": "mock://test.com"}).create_request_object()
+
+    def test_method(self):
+        assert HTTPHelperRequestHandler({"method": "GET"}).get_request_model().method == "GET"
+        assert HTTPHelperRequestHandler({"method": "RANDOM"}).get_request_model().method == "RANDOM"
+        assert HTTPHelperRequestHandler({"method": "GET"}).validate_method() is True
+
+        try:
+            HTTPHelperRequestHandler().validate_method()
+            assert False
+        except ValueError:
+            assert True
+
+        try:
+            assert HTTPHelperRequestHandler({
+                "method": "RANDOM"
+            }).validate_method()
+            assert False
+
+        except NotImplementedError:
+            assert True
+
+    def test_url(self):
+        assert HTTPHelperRequestHandler({"endpoint": "http://mock.com"}).get_request_model().endpoint == \
+               "http://mock.com"
+        assert HTTPHelperRequestHandler({"endpoint": "http://mock.com"}).validate_endpoint() is True
+
+        try:
+            HTTPHelperRequestHandler().validate_endpoint()
+            assert False
+        except ValueError:
+            assert True
+
+        try:
+            HTTPHelperRequestHandler(-1).validate_endpoint()
+            assert False
```

Comment:
https://docs.pytest.org/en/stable/reference.html#pytest-fail is how you fail a test. `assert False` will just produce a meaningless result. Writing a failure message gives you a chance to explain to the reader of both your code and the failed test what you had expected.

Reply:
Replaced with pytest